Source/OxyPlot/OxyPlot.csproj

```diff
@@ -1,26 +1,17 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
     <PropertyGroup>
-        <TargetFrameworks>netstandard1.0;netstandard2.0;net462</TargetFrameworks>
+        <TargetFrameworks>netstandard2.0;net462;net6.0;net7.0</TargetFrameworks>
         <PackageId>OxyPlot.Core</PackageId>
         <Description>OxyPlot is a plotting library for .NET. This is the core library including the foundation classes and plot model. To display the plots, you also need to add a platform-specific OxyPlot package that contains a PlotView component.</Description>
-        <Copyright>Copyright (c) 2014 OxyPlot contributors</Copyright>
-        <PackageLicenseExpression>MIT</PackageLicenseExpression>
-        <PackageProjectUrl>https://oxyplot.github.io/</PackageProjectUrl>
-        <PackageIcon>OxyPlot_128.png</PackageIcon>
         <PackageTags>plotting plot charting chart</PackageTags>
-        <RepositoryType>git</RepositoryType>
-        <RepositoryUrl>https://github.com/oxyplot/oxyplot.git</RepositoryUrl>
-        <GenerateDocumentationFile>True</GenerateDocumentationFile>
+        <GenerateDocumentationFile>true</GenerateDocumentationFile>
         <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-        <SignAssembly>True</SignAssembly>
+        <SignAssembly>true</SignAssembly>
         <AssemblyOriginatorKeyFile>OxyPlot.snk</AssemblyOriginatorKeyFile>
         <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
         <LangVersion>8</LangVersion>
-        <NoWarn>618</NoWarn>
+        <NoWarn>$(NoWarn);CS0618;CS8767</NoWarn>
```

Comment:
Would prefer we disable CS8767 only in the one place it lights up (`CompareHelper`) with `#pragma warning disable/restore CS8767`. Without changing to C#9 (which I don't think is supported on net framework) I can't find a way to make the code happy.

Reply:
> Without changing to C#9 (which I don't think is supported on net framework) I can't find a way to make the code happy.

.Net Framework supports C# 9. You can also always use PolySharp - https://github.com/Sergio0694/PolySharp

Reply:
@HavenDV do you have a reference for C#9 support in net framework? I searched but could only find comments about net 5.0 and above, which I took to mean net framework, but keen to hear otherwise.

Reply:
I have many projects that include net451 or net462 TargetFramework and for which I install the latest version of the language. Everything works correctly. But, for example, to support records, you need to add special classes to the project (or use PolySharp, which will do the same)

Reply:
This states it is not supported (in 4.8 at least), but I still can't find a clear definitive reference https://github.com/dotnet/docs/issues/26587#issuecomment-947693187

Reply:
Not everything on the internet is correct.
Probably the easiest way, if you don't believe me, is to ask the community of other C# developers
https://discord.gg/uwp - there are many developers here, including those from Microsoft. Despite the name, it's not just about UWP

Reply:
I can confirm this, I always use the latest version of C# even on a project where I am stuck on .NET Framework. Some stuff works right out of the box, for some stuff you need to supply the runtime features (didn't know about PolySharp, thanks for the link!), and some stuff doesn't work at all. But, importantly, it doesn't break the build, even if not all features are supported.
I think we could consider bumping our `LangVersion` to 11. Doesn't have to be part of this PR though (but can, if you are motivated @mattico).

Reply:
I'm tempted to suggest we remove the `ComparisonHelper` API and rewrite the only usage thereof (which is confusing and depends on implementation details in `List<T>.BinarySearch`). For the time being, it's probably simplest just to disable the warning locally, which won't create a burden anywhere else.

Reply:
Quoting the [MSDN](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version#:~:text=Choosing%20a%20language%20version%20newer%20than%20the%20default%20can%20cause%20hard%20to%20diagnose%20compile%2Dtime%20and%20runtime%20errors.)
> Choosing a language version newer than the default can cause hard to diagnose compile-time and runtime errors.

Certainly OxyPlot isn't doing anything exciting run-time wise, and I'm not aware of any situations where it actually goes wrong, but I'd still be uneasy using something that isn't officially supported.

Reply:
With our `LangVersion` at 8 we are actually already in unsupported territory. I suspect MS writes this as they want a) people to move away from .NET Framework and b) don't want to be held responsible for any misuses.

But I agree, let's keep it simple for now, the `LangVersion` might be a discussion for another day. Let's disable the warning locally and keep everything else as is.

Reply:
The question for me regarding bumping the language version is really whether it can create silent bugs. Continuing the search, I can't find any evidence that this has occurred, and most discussion is just concerning using newer features.

All that said, it looks like C# 8 isn't strictly supported either, so I should probably stop worrying.

Reply:
(Sorry for the confusing messages; I've managed to miss both of @Jonarw 's message before sending mine somehow)